task/mypy-lint/0.2/mypy-lint.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mypy-lint
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Code Quality
+    tekton.dev/tags: linter, mypy
+    tekton.dev/displayName: "Mypy(Python) linter"
+spec:
+  description: >-
+    This task can be used to perform lint check on python files
+  workspaces:
+    - name: shared-workspace
+      description: A workspace containing files that we want to check
+  params:
+    - name: image
+      description: The container image to use mypy in
+      default: docker.io/cytopia/mypy:latest-0.4@sha256:0d5e1a97ce68c4aeececc818c8bf19301b396edbe36a6f83eca8248d7d84a05c
+    - name: requirements_file
+      description: The name of the requirements file inside the source location
+      default: requirements.txt
+    - name: args
+      type: array
+      description: extra args that need to be appended
+      default: ["--help"]
+  steps:
+    - name: lint-python-files
+      image: $(params.image)
+      workingDir: $(workspaces.shared-workspace.path)
+      script: |
+        export HOME=/tmp/python
```

Comment:
should you shomehow mkdir -p it ? i guess mypy does it by itself?

Reply:
To be completely honest, I wasn't really sure what that part of the code did, I just copied it from https://github.com/tektoncd/catalog/blob/main/task/pylint/0.2/pylint.yaml 

So, to find out I spun up a container and checked 
```
/data # export HOME=/tmp/python
/data # export PATH=$PATH:/tmp/python/.local/bin
/data # ls /tmp/python
ls: /tmp/python: No such file or directory
/data # echo mypy > requirements.txt
/data # python -mpip install --user -r requirements.txt
Requirement already satisfied: mypy in /usr/lib/python3.8/site-packages (from -r requirements.txt (line 1)) (0.901)
Requirement already satisfied: mypy-extensions<0.5.0,>=0.4.3 in /usr/lib/python3.8/site-packages (from mypy->-r requirements.txt (line 1)) (0.4.3)
Requirement already satisfied: typing-extensions>=3.7.4 in /usr/lib/python3.8/site-packages (from mypy->-r requirements.txt (line 1)) (3.10.0.0)
Requirement already satisfied: toml in /usr/lib/python3.8/site-packages (from mypy->-r requirements.txt (line 1)) (0.10.2)
WARNING: Running pip as root will break packages and permissions. You should install packages reliably by using venv: https://pip.pypa.io/warnings/venv
WARNING: You are using pip version 21.1.2; however, version 21.2.3 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
/data # ls -lthr /tmp/python/.local/bin
total 20K
-rwxr-xr-x    1 root     root        1.4K Aug 10 12:03 mypyc
-rwxr-xr-x    1 root     root         217 Aug 10 12:03 stubtest
-rwxr-xr-x    1 root     root         216 Aug 10 12:03 stubgen
-rwxr-xr-x    1 root     root         235 Aug 10 12:03 mypy
-rwxr-xr-x    1 root     root         239 Aug 10 12:03 dmypy
```

So it seems this will be created by the `pip install` command if it does not exist yet.

Reply:
great thanks!